Demo/src/main/java/com/paypal/android/ui/approveorder/ApproveOrderViewModel.kt

```diff
@@ -99,23 +98,17 @@ class ApproveOrderViewModel @Inject constructor(
 
     fun completeOrder(context: Context) {
         viewModelScope.launch {
-            isCompleteOrderLoading = true
+            completeOrderState = ActionState.Loading
 
             val cmid = payPalDataCollector.collectDeviceData(context)
-            val orderId = createdOrder!!.id!!
-
-            completedOrder = completeOrderUseCase(orderId, intentOption, cmid)
-            isCompleteOrderLoading = false
+            val completedOrder = completeOrderUseCase(createdOrder!!.id!!, intentOption, cmid)
```

Comment:
Yeah great call. The best solution will most likely be to display an error in this case. It's tough because if it gets to that point it's highly probably that `createdOrder` is non-null. It makes me wonder if it's a code smell? I wonder if there's a design pattern we can leverage here to make it super clean.

Reply:
One option would be to fail fast when a merchant launches the SDK and required params or config is missing. This would allow us to have non-null types for these fields inside the SDK. No need to update this PR, but it might be someone looking into as a tech debt story or hack day project.

Reply:
Ok cool. That does sound like a good idea. We can probably return an error result immediately from the UseCase to prevent the force unwraps.

Reply:
I'm gonna create a ticket for this tech debt story. We can fix this for all `UseCase` related code to make sure we're handling errors properly throughout the Demo app.

Reply:
Added a ticket DTPPCPSDK-1706 for addressing the tech debt.